io/plugins/eu.esdihumboldt.hale.io.shp/src/eu/esdihumboldt/hale/io/shp/writer/ShapefileInstanceWriter.java

```diff
@@ -112,10 +116,18 @@ protected IOReport execute(ProgressIndicator progress, IOReporter reporter)
 			filesWritten = writeInstances(instances, progress, reporter, location);
 
 			if (filesWritten.size() > 1) {
-				List<URI> uris = filesWritten.stream().map(f -> {
+
+				List<URI> uris = new ArrayList<>();
+
+				for (String f : filesWritten) {
 					File file = new File(filePath + "/" + f + ShapefileConstants.SHP_EXTENSION);
-					return file.toURI();
-				}).collect(Collectors.toList());
+					uris.add(file.toURI());
+
+					String cpgFileName = filePath + "/" + f + ShapefileConstants.CPG_EXTENSION;
+					writeCodePageFile(cpgFileName);
+					file = new File(cpgFileName);
+					uris.add(file.toURI());
```

Comment:
This will contain [name of the file](https://github.com/halestudio/hale/blob/a970f6e19cac38578fb00eb75baa31808df22423/io/plugins/eu.esdihumboldt.hale.io.shp/src/eu/esdihumboldt/hale/io/shp/writer/ShapefileInstanceWriter.java#L693C8-L693C8) without extension.